diff --git a/examples/companion_radio/MyMesh.cpp b/examples/companion_radio/MyMesh.cpp
index 3772fc1..43f2a09 100644
--- a/examples/companion_radio/MyMesh.cpp
+++ b/examples/companion_radio/MyMesh.cpp
@@ -234,7 +234,22 @@ uint8_t MyMesh::getExtraAckTransmitCount() const {
   return _prefs.multi_acks;
 }
 
+const char *MyMesh::getLogDateTime() {
+  static char tmp[32];
+  uint32_t now = getRTCClock()->getCurrentTime();
+  DateTime dt = DateTime(now);
+  sprintf(tmp, "%02d:%02d:%02d - %d/%d/%d U", dt.hour(), dt.minute(), dt.second(), dt.day(), dt.month(),
+          dt.year());
+  return tmp;
+}
+
 void MyMesh::logRxRaw(float snr, float rssi, const uint8_t raw[], int len) {
+#if MESH_PACKET_LOGGING
+  Serial.print(getLogDateTime());
+  Serial.print(" RAW: ");
+  mesh::Utils::printHex(Serial, raw, len);
+  Serial.println();
+#endif
   if (_serial->isConnected() && len + 3 <= MAX_FRAME_SIZE) {
     int i = 0;
     out_frame[i++] = PUSH_CODE_LOG_RX_DATA;
diff --git a/examples/companion_radio/MyMesh.h b/examples/companion_radio/MyMesh.h
index e640087..ba10dc3 100644
--- a/examples/companion_radio/MyMesh.h
+++ b/examples/companion_radio/MyMesh.h
@@ -94,6 +94,8 @@ public:
   NodePrefs *getNodePrefs();
   uint32_t getBLEPin();
 
+  const char* getLogDateTime();
+
   void loop();
   void handleCmdFrame(size_t len);
   bool advert();
diff --git a/variants/heltec_v4/platformio.ini b/variants/heltec_v4/platformio.ini
index 2c06b97..45b95b4 100644
--- a/variants/heltec_v4/platformio.ini
+++ b/variants/heltec_v4/platformio.ini
@@ -153,7 +153,7 @@ build_flags =
   -D AUTO_SHUTDOWN_MILLIVOLTS=3400
   -D BLE_DEBUG_LOGGING=1
   -D OFFLINE_QUEUE_SIZE=256
-;  -D MESH_PACKET_LOGGING=1
+  -D MESH_PACKET_LOGGING=1
 ;  -D MESH_DEBUG=1
 build_src_filter = ${Heltec_lora32_v4.build_src_filter}
   +<helpers/ui/SSD1306Display.cpp>
