diff --git a/examples/companion_radio/main.cpp b/examples/companion_radio/main.cpp
index bb82742..177f22e 100644
--- a/examples/companion_radio/main.cpp
+++ b/examples/companion_radio/main.cpp
@@ -33,7 +33,7 @@
   #define LORA_CR      5
 #endif
 #ifndef LORA_TX_POWER
-  #define LORA_TX_POWER  20
+  #define LORA_TX_POWER  27
 #endif
 #ifndef MAX_LORA_TX_POWER
   #define MAX_LORA_TX_POWER  LORA_TX_POWER
diff --git a/examples/simple_repeater/main.cpp b/examples/simple_repeater/main.cpp
index 957892b..bbbfef2 100644
--- a/examples/simple_repeater/main.cpp
+++ b/examples/simple_repeater/main.cpp
@@ -43,7 +43,7 @@
   #define LORA_CR      5
 #endif
 #ifndef LORA_TX_POWER
-  #define LORA_TX_POWER  20
+  #define LORA_TX_POWER  27
 #endif
 
 #ifndef ADVERT_NAME
diff --git a/examples/simple_room_server/main.cpp b/examples/simple_room_server/main.cpp
index 25ec3a1..dc0921b 100644
--- a/examples/simple_room_server/main.cpp
+++ b/examples/simple_room_server/main.cpp
@@ -43,7 +43,7 @@
   #define LORA_CR      5
 #endif
 #ifndef LORA_TX_POWER
-  #define LORA_TX_POWER  20
+  #define LORA_TX_POWER  27
 #endif
 
 #ifndef ADVERT_NAME
diff --git a/examples/simple_secure_chat/main.cpp b/examples/simple_secure_chat/main.cpp
index 4eae2e3..41b7be6 100644
--- a/examples/simple_secure_chat/main.cpp
+++ b/examples/simple_secure_chat/main.cpp
@@ -34,7 +34,7 @@
   #define LORA_CR      5
 #endif
 #ifndef LORA_TX_POWER
-  #define LORA_TX_POWER  20
+  #define LORA_TX_POWER  27
 #endif
 
 #ifndef MAX_CONTACTS
diff --git a/variants/heltec_v2/platformio.ini b/variants/heltec_v2/platformio.ini
index cfca7fe..5d74484 100644
--- a/variants/heltec_v2/platformio.ini
+++ b/variants/heltec_v2/platformio.ini
@@ -11,7 +11,7 @@ build_flags =
   -D PIN_OLED_RESET=16
   -D RADIO_CLASS=CustomSX1276
   -D WRAPPER_CLASS=CustomSX1276Wrapper
-  -D LORA_TX_POWER=20
+  -D LORA_TX_POWER=27
   -D P_LORA_TX_LED=25
 build_src_filter = ${esp32_base.build_src_filter}
   +<../variants/heltec_v2>
diff --git a/variants/heltec_v3/platformio.ini b/variants/heltec_v3/platformio.ini
index cfe8f26..03f071f 100644
--- a/variants/heltec_v3/platformio.ini
+++ b/variants/heltec_v3/platformio.ini
@@ -7,7 +7,7 @@ build_flags =
   -D HELTEC_LORA_V3
   -D RADIO_CLASS=CustomSX1262
   -D WRAPPER_CLASS=CustomSX1262Wrapper
-  -D LORA_TX_POWER=22
+  -D LORA_TX_POWER=27
   -D P_LORA_TX_LED=35
   -D PIN_BOARD_SDA=17
   -D PIN_BOARD_SCL=18
diff --git a/variants/lilygo_t3s3/platformio.ini b/variants/lilygo_t3s3/platformio.ini
index 59ed60b..35f3111 100644
--- a/variants/lilygo_t3s3/platformio.ini
+++ b/variants/lilygo_t3s3/platformio.ini
@@ -24,7 +24,7 @@ build_flags =
   -D SX126X_CURRENT_LIMIT=130
   -D RADIO_CLASS=CustomSX1262
   -D WRAPPER_CLASS=CustomSX1262Wrapper
-  -D LORA_TX_POWER=22
+  -D LORA_TX_POWER=27
   -D SX126X_RX_BOOSTED_GAIN=1
 build_src_filter = ${esp32_base.build_src_filter}
   +<../variants/lilygo_t3s3>
diff --git a/variants/lilygo_tbeam/platformio.ini b/variants/lilygo_tbeam/platformio.ini
index f754896..d609e0a 100644
--- a/variants/lilygo_tbeam/platformio.ini
+++ b/variants/lilygo_tbeam/platformio.ini
@@ -7,7 +7,7 @@ build_flags =
   -D LILYGO_TBEAM
   -D RADIO_CLASS=CustomSX1276
   -D WRAPPER_CLASS=CustomSX1276Wrapper
-  -D LORA_TX_POWER=20
+  -D LORA_TX_POWER=27
   -D P_LORA_TX_LED=4
   -D PIN_BOARD_SDA=21
   -D PIN_BOARD_SCL=22
diff --git a/variants/lilygo_tlora_v2_1/platformio.ini b/variants/lilygo_tlora_v2_1/platformio.ini
index f7485f9..4c04bb1 100644
--- a/variants/lilygo_tlora_v2_1/platformio.ini
+++ b/variants/lilygo_tlora_v2_1/platformio.ini
@@ -23,7 +23,7 @@ build_flags =
   -D ARDUINO_LOOP_STACK_SIZE=16384
   -D DISPLAY_CLASS=SSD1306Display
   -D WRAPPER_CLASS=CustomSX1276Wrapper
-  -D LORA_TX_POWER=20
+  -D LORA_TX_POWER=27
 build_src_filter = ${esp32_base.build_src_filter}
   +<../variants/lilygo_tlora_v2_1>
 lib_deps =
diff --git a/variants/promicro/platformio.ini b/variants/promicro/platformio.ini
index f99a427..8d7d7bb 100644
--- a/variants/promicro/platformio.ini
+++ b/variants/promicro/platformio.ini
@@ -6,7 +6,7 @@ build_flags = ${nrf52840_base.build_flags}
   -D FAKETEC
   -D RADIO_CLASS=CustomSX1262
   -D WRAPPER_CLASS=CustomSX1262Wrapper
-  -D LORA_TX_POWER=22
+  -D LORA_TX_POWER=27
   -D SX126X_CURRENT_LIMIT=130
   -D SX126X_RX_BOOSTED_GAIN=1
   -D DISPLAY_CLASS=SSD1306Display
@@ -106,7 +106,7 @@ build_flags = ${nrf52840_base.build_flags}
   -D FAKETEC
   -D RADIO_CLASS=CustomLLCC68 
   -D WRAPPER_CLASS=CustomLLCC68Wrapper
-  -D LORA_TX_POWER=22
+  -D LORA_TX_POWER=27
   -D SX126X_CURRENT_LIMIT=130
   -D SX126X_RX_BOOSTED_GAIN=1
 build_src_filter = ${nrf52840_base.build_src_filter}
diff --git a/variants/rak4631/platformio.ini b/variants/rak4631/platformio.ini
index 518ab8b..22ca26f 100644
--- a/variants/rak4631/platformio.ini
+++ b/variants/rak4631/platformio.ini
@@ -9,7 +9,7 @@ build_flags = ${nrf52840_base.build_flags}
   -D PIN_USER_BTN=9
   -D RADIO_CLASS=CustomSX1262
   -D WRAPPER_CLASS=CustomSX1262Wrapper
-  -D LORA_TX_POWER=22
+  -D LORA_TX_POWER=27
   -D SX126X_CURRENT_LIMIT=130
   -D SX126X_RX_BOOSTED_GAIN=1
 build_src_filter = ${nrf52840_base.build_src_filter}
diff --git a/variants/station_g2/platformio.ini b/variants/station_g2/platformio.ini
index 96eca1c..9b88594 100644
--- a/variants/station_g2/platformio.ini
+++ b/variants/station_g2/platformio.ini
@@ -7,7 +7,7 @@ build_flags =
   -D STATION_G2
   -D RADIO_CLASS=CustomSX1262
   -D WRAPPER_CLASS=CustomSX1262Wrapper
-  -D LORA_TX_POWER=7
+  -D LORA_TX_POWER=27
 ;  -D P_LORA_TX_LED=35
 ;  -D PIN_BOARD_SDA=5
 ;  -D PIN_BOARD_SCL=6
diff --git a/variants/t1000-e/platformio.ini b/variants/t1000-e/platformio.ini
index e5c0d11..3737ab8 100644
--- a/variants/t1000-e/platformio.ini
+++ b/variants/t1000-e/platformio.ini
@@ -26,7 +26,7 @@ build_flags = ${nrf52840_t1000e.build_flags}
   -D PIN_STATUS_LED=24
   -D RADIO_CLASS=CustomLR1110
   -D WRAPPER_CLASS=CustomLR1110Wrapper
-  -D LORA_TX_POWER=22
+  -D LORA_TX_POWER=27
 build_src_filter = ${nrf52840_t1000e.build_src_filter}
   +<helpers/*.cpp>
   +<helpers/nrf52/T1000eBoard.cpp>
diff --git a/variants/t114/platformio.ini b/variants/t114/platformio.ini
index 754839d..98d963a 100644
--- a/variants/t114/platformio.ini
+++ b/variants/t114/platformio.ini
@@ -19,7 +19,7 @@ build_flags = ${nrf52840_t114.build_flags}
   -D P_LORA_TX_LED=35
   -D RADIO_CLASS=CustomSX1262
   -D WRAPPER_CLASS=CustomSX1262Wrapper
-  -D LORA_TX_POWER=22
+  -D LORA_TX_POWER=27
   -D SX126X_CURRENT_LIMIT=130
   -D SX126X_RX_BOOSTED_GAIN=1
 build_src_filter = ${nrf52840_t114.build_src_filter}
diff --git a/variants/techo/platformio.ini b/variants/techo/platformio.ini
index 2c825e6..a82d575 100644
--- a/variants/techo/platformio.ini
+++ b/variants/techo/platformio.ini
@@ -18,7 +18,7 @@ build_flags = ${nrf52840_techo.build_flags}
   -DLILYGO_TECHO
   -D RADIO_CLASS=CustomSX1262
   -D WRAPPER_CLASS=CustomSX1262Wrapper
-  -D LORA_TX_POWER=22
+  -D LORA_TX_POWER=27
   -D SX126X_CURRENT_LIMIT=130
   -D SX126X_RX_BOOSTED_GAIN=1
 build_src_filter = ${nrf52840_techo.build_src_filter}
diff --git a/variants/xiao_c3/platformio.ini b/variants/xiao_c3/platformio.ini
index 77c9eae..8bdf5c6 100644
--- a/variants/xiao_c3/platformio.ini
+++ b/variants/xiao_c3/platformio.ini
@@ -28,7 +28,7 @@ build_flags =
   -D RADIO_CLASS=CustomSX1262
   -D WRAPPER_CLASS=CustomSX1262Wrapper
   -D SX126X_RX_BOOSTED_GAIN=1
-  -D LORA_TX_POWER=22
+  -D LORA_TX_POWER=27
   -D ADVERT_NAME='"Xiao Repeater"'
   -D ADVERT_LAT=-37.0
   -D ADVERT_LON=145.0
@@ -47,7 +47,7 @@ build_flags =
   ${Xiao_esp32_C3.build_flags}
   -D RADIO_CLASS=CustomSX1268
   -D WRAPPER_CLASS=CustomSX1268Wrapper
-  -D LORA_TX_POWER=22
+  -D LORA_TX_POWER=27
   -D ADVERT_NAME='"Xiao Repeater"'
   -D ADVERT_LAT=-37.0
   -D ADVERT_LON=145.0
diff --git a/variants/xiao_s3_wio/platformio.ini b/variants/xiao_s3_wio/platformio.ini
index 9eedb4f..63f4e58 100644
--- a/variants/xiao_s3_wio/platformio.ini
+++ b/variants/xiao_s3_wio/platformio.ini
@@ -18,7 +18,7 @@ build_flags = ${esp32_base.build_flags}
   -D SX126X_CURRENT_LIMIT=130
   -D RADIO_CLASS=CustomSX1262
   -D WRAPPER_CLASS=CustomSX1262Wrapper
-  -D LORA_TX_POWER=22
+  -D LORA_TX_POWER=27
   -D SX126X_RX_BOOSTED_GAIN=1
 build_src_filter = ${esp32_base.build_src_filter}
   +<../variants/xiao_s3_wio>
